source/Irrlicht/CMakeLists.txt

```diff
@@ -406,6 +374,23 @@ add_library(IRRVIDEOOBJ OBJECT
 	${IRRIMAGEOBJ}
 )
 
+target_link_libraries(IRRVIDEOOBJ
+	PUBLIC
+		# TODO Create imported targets if they do not already exist,
+		# and only link the correct one for the enabled backend.
+		"$<$<BOOL:${OPENGL_DIRECT_LINK}>:${OPENGL_LIBRARIES}>"
+		"$<$<BOOL:${OPENGLES_DIRECT_LINK}>:${OPENGLES_LIBRARY}>"
+		"$<$<BOOL:${OPENGLES2_DIRECT_LINK}>:${OPENGLES2_LIBRARIES}>"
+		"${EGL_LIBRARY}"
+		JPEG::JPEG
+	PRIVATE
+		PNG::PNG
+)
+
+if(USE_X11)
+	target_include_directories(IRRVIDEOOBJ PUBLIC "${X11_INCLUDE_DIR}")
```

Comment:
this could be a generator expression instead

Reply:
That will be obsolete, though, once I change it to use a target.